### PR TITLE
Fix iOS contact hashing

### LINKED ISSUE #6776: iOS Contact Synchronisation creates duplicates each time Tuta Mail gets launched
- [x] This is not a feature request (existing functionality does not work, **not** missing functionality).
 I will request features on [forum](https://www.reddit.com/r/tutanota/) or via support.
- [x] I've searched and did not find a similar issue.

**Bug in mobile app**
Yes

**Describe the bug**
I have created one contact in the Tuta Mail Adressbook called "Test Contact".
On the iPhone, when I launch Tuta Mail and login, the contact gets added to the iPhone Contact App.
But it does so every time Tuta Mail gets launched. Instead of maybe overwriting the old "Test Contact" it creates a duplicate.
This happens each time I launch the App. Meaning that after 10 launches I have 10 "Test Contact" contacts.

If I leave Tuta Mail running in the background and switch back to it, no duplicates get created.
It only happens when closing and reopening the app.

I have only one iOS device with contact synchronisation. No other devices.

**To Reproduce**
1. Create contact in Tuta Mail (does not matter if on phone, desktop or web)
2. Open iOS Tuta Mail App
3. Check Contacts App -> 1 Contact created
4. Close Tuta Mail
5. Launch Tuta Mail
6. Check Contacts App -> Second entry with same name got created

**Expected behavior**
The contact should only be created once.
Per description in the Tuta Mail App Settings page, the "old" contact should get overwritten.

**Screenshots**
image1
image0

**Smartphone (please complete the following information):**
 - Device: iPhone 15 Pro
 - OS: iOS 17.4.1
 - Tuta Mail Version: 218.240307.1

### PR DESCRIPTION
**Pull request by murilopereirame**
The original contact hashing used to check contact changes was based on Swift's implementation of Hashable that unfortunately is granted to be the same just during the program execution, leading to different hashes every app startup.

This commit replaces the use of Hashble by an implementation of MurmurHash, leading to stable hashes across multiples app startup.

Co-authored-by: ivk <ivk@tutao.de>

fix #6776

### CODE REVIEW

File: app-ios/tutanota/Sources/Contacts/StructuredContactTypes.swift [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import Contacts
+import CryptoKit
```
Comment (wrdhub) [inline app-ios/tutanota/Sources/Contacts/StructuredContactTypes.swift:2]: Is this import still needed?

File: app-ios/tutanota/Sources/Utils/MurMurHash.swift [hunk 2]
```diff
@@ -0,0 +1,304 @@
+//
+//  MurmurHash3.swift
+//  MurmurHash
+//
+//  Created by Daisuke T on 2019/02/06.
+//  Copyright © 2019 MurmurHash. All rights reserved.
```
Comment (rezbyte) [inline app-ios/tutanota/Sources/Utils/MurMurHash.swift:6]: Do we need to [provide the license for `MurmurHash-Swift`](https://github.com/daisuke-t-jp/MurmurHash-Swift/blob/master/LICENSE) anywhere?

File: app-ios/.swiftlint.yml [hunk 3]
```diff
@@ -41,7 +41,6 @@ opt_in_rules:
 - discouraged_object_literal
 - empty_collection_literal
 - empty_string
-- fallthrough
```
Comment (rezbyte) [inline app-ios/.swiftlint.yml:44]: Was silencing the rule for `MurMurHash` not an option? Removing an entire rule just for one PR seems a bit extreme.

File: app-ios/tutanota/Sources/Utils/MurMurHash.swift (hunk 2 above)
Comment (rezbyte) [reply app-ios/tutanota/Sources/Utils/MurMurHash.swift:6]: As far as I know legal-wise, that covers it. :+1:

File: app-ios/.swiftlint.yml (hunk 3 above)
Comment (rezbyte) [reply app-ios/.swiftlint.yml]: Maybe we could start a poll somewhere to vote on what linting rules to remove. There seems to be a lot of complaints about linting rules.

**Pull request merged**